lib/bitcoin/BitcoinProcessor.ts

```diff
@@ -176,8 +191,144 @@ export default class BitcoinProcessor {
   }
 
   /**
-   * Trims the databases to the block prior to the block the given transaction is in.
-   * @returns The last fully processed block after trimming. `undefined` if all data are deleted after trimming.
+   * A faster version of process transactions that requires access to bitcoin data directory
+   * @param startingBlock the starting block to begin processing
+   */
+  private async fastProcessTransactions (startingBlock: IBlockInfo) {
+    const bitcoinBlockDataIterator = new BitcoinBlockDataIterator(this.bitcoinDataDirectory!);
+    const bestHeight = await this.bitcoinClient.getCurrentBlockHeight();
+    const bestHash = await this.bitcoinClient.getBlockHash(bestHeight);
+
+    let currentHash = bestHash;
+    let currentHeight = bestHeight;
+    const numOfBlocksToProcess = bestHeight - startingBlock.height + 1;
+
+    // An array of maps<hash, IBlockInfoExtended>.
+    const chainInfos = [...new Array(numOfBlocksToProcess)].map<Map<string, IBlockInfoExtended>>(() => new Map());
+
+    console.log(`Begin fast processing block ${startingBlock.height} to ${bestHeight}`);
+    // loop through files and process them until we process all blocks needed
+    while (bitcoinBlockDataIterator.hasNext() && currentHeight >= startingBlock.height) {
+      const blockData = bitcoinBlockDataIterator.next()!;
+      await this.processBlockData(blockData, chainInfos, startingBlock.height, currentHeight);
+      [currentHeight, currentHash] = await this.markBestHeight(chainInfos, currentHeight, currentHash, startingBlock.height);
+    }
+
+    this.lastProcessedBlock = {
+      height: bestHeight,
+      hash: bestHash,
+      previousHash: chainInfos[chainInfos.length - 1]!.get(bestHash)!.previousHash
+    };
+    console.log('finished fast processing');
+    // TODO: Issue #783
+    // Need to use the fee infos and set fee after here.
+  }
+
+  private async processBlockData (
+    blockData: {[blockHash: string]: BitcoinBlockModel},
+    chainInfos: Map<string, IBlockInfoExtended>[],
+    startingBlockHeight: number,
+    currentHeight: number) {
+
+    for (let blockHash in blockData) {
+      const block = blockData[blockHash];
```

Comment:
This feels like a round about way to iterate through block data, why not just return a map, or array directly from the parser for `blockData`?

Reply:
Will change to Array after current review.